tensorflow/core/kernels/data/experimental/sampling_dataset_op_test.cc

```diff
@@ -0,0 +1,847 @@
+    if (input_dataset.NumElements() == 0 ||
+        input_dataset.dtype() != DT_VARIANT) {
+      return tensorflow::errors::Internal(
+          "The input dataset is not populated as the dataset tensor yet.");
+    }
+    *inputs = {TensorValue(&input_dataset), TensorValue(&rate),
+               TensorValue(&seed), TensorValue(&seed2)};
+    return Status::OK();
+  }
+
+  // Static parameters of the kernel
+  Tensor rate;
+  Tensor seed;
+  Tensor seed2;
+
+  // Parameters of the sequence of numbers that will serve as the dynamic input
+  // of the kernel.
+  LocalRangeDatasetParams range_dataset_params;
+
+  // RangeDataset kernel wrapped in a variant tensor. Initialized by the test
+  // case itself because the MakeRangeDataset() method requires an instance of
+  // DatasetOpsTestBase.
+  Tensor input_dataset;
+};
+
+SamplingDatasetParams OneHundredPercentSampleDataset() {
+  return {/*rate*/ 1.0,
+          /*seed*/ 42,
+          /*seed2*/ 7,
+          /*start*/ 0,
+          /*stop*/ 3,
+          /*step*/ 1,
+          /*output_dtypes*/ {DT_INT64},
+          /*output_shapes*/ {PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+SamplingDatasetParams TenPercentSampleDataset() {
+  return {/*rate*/ 0.1,
+          /*seed*/ 42,
+          /*seed2*/ 7,
+          /*start*/ 0,
+          /*stop*/ 20,
+          /*step*/ 1,
+          /*output_dtypes*/ {DT_INT64},
+          /*output_shapes*/ {PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+SamplingDatasetParams ZeroPercentSampleDataset() {
+  return {/*rate*/ 0.0,
+          /*seed*/ 42,
+          /*seed2*/ 7,
+          /*start*/ 0,
+          /*stop*/ 20,
+          /*step*/ 1,
+          /*output_dtypes*/ {DT_INT64},
+          /*output_shapes*/ {PartialTensorShape({})},
+          /*node_name=*/kNodeName};
+}
+
+class ParameterizedGetNextSamplingDatasetOpTest
+    : public SamplingDatasetOpTest,
+      public ::testing::WithParamInterface<
+          GetNextTestCase<SamplingDatasetParams>> {};
+
+// Test case 1: 100% sample should return all inputs
+GetNextTestCase<SamplingDatasetParams> GetNextTestCase1() {
+  return {/*dataset_params=*/OneHundredPercentSampleDataset(),
+          /*expected_outputs=*/
+          CreateTensors<int64>(TensorShape({}), {{0}, {1}, {2}})};
+}
+
+// Test case 2: 10% sample should return about 10% of inputs, and the specific
+// inputs returned shouldn't change across build environments.
+GetNextTestCase<SamplingDatasetParams> GetNextTestCase2() {
+  return {/*dataset_params=*/TenPercentSampleDataset(),
+          /*expected_outputs=*/
+          CreateTensors<int64>(TensorShape({}), {{9}, {11}, {19}})};
+}
+
+// Test case 3: 0% sample should return nothing and should not crash.
+GetNextTestCase<SamplingDatasetParams> GetNextTestCase3() {
+  return {/*dataset_params=*/ZeroPercentSampleDataset(),
+          /*expected_outputs=*/{}};
+}
+
+TEST_P(ParameterizedGetNextSamplingDatasetOpTest, GetNext) {
```

Comment:
I believe that @feihugis is working on utilities for factoring out the commonalities in test setup. Fei, would you refactoring be able to avoid the duplication in code in the tests here?

Reply:
@jsimsa Yes, the refactoring I'm working on will avoid the duplication here. Once the solution is finalized, I will work with @frreiss to remove the duplications here.

Reply:
I've removed the duplicate initialization code as part of the refactoring in commit d827bfa.